src/main/java/gregtech/api/util/OverlayedItemHandler.java

```diff
@@ -49,20 +49,20 @@ private void initSlot(int slot) {
     }
 
 
-    public int insertStackedItemStackKey(ItemStackKey key, int amountToInsert) {
+    public int insertStackedItemStackKey(ItemStack key, int amountToInsert) {
         int lastKnownPopulatedSlot = 0;
         //loop through all slots, looking for ones matching the key
         for (int i = 0; i < this.slots.length; i++) {
             //populate the slot if it's not already populated
             initSlot(i);
             // if it's the same item or there is no item in the slot
-            ItemStackKey slotKey = this.slots[i].getItemStackKey();
-            if (slotKey == key || slotKey == null) {
-                //if the slot its not full
-                int canInsertUpTo = this.slots[i].slotLimit - this.slots[i].count;
+            ItemStack slotKey = this.slots[i].getItemStack();
+            if (slotKey.isEmpty() || ItemStackHashStrategy.comparingAllButCount().equals(slotKey, key)) {
+                //if the slot is not full
+                int canInsertUpTo = this.slots[i].getSlotLimit() - this.slots[i].getCount();
                 if (canInsertUpTo > 0) {
                     int insertedAmount = Math.min(canInsertUpTo, amountToInsert);
-                    this.slots[i].setItemStackKey(key);
+                    this.slots[i].setItemStack(key);
```

Comment:
I'm not sure about this. Removing the copy does not affect the results of our current tests, but I can't determine if this will cause issues in production.